creator/data_templates/mutations/template_version.py

```diff
@@ -0,0 +1,259 @@
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.db import transaction
+from django.conf import settings
+
+from creator.studies.models import Study
+from creator.data_templates.models import (
+    DataTemplate,
+    TemplateVersion,
+)
+from creator.data_templates.nodes.template_version import TemplateVersionNode
+
+
+def check_studies(study_node_ids):
+    """
+    Check if studies exist and return the Study objects if all studies exist
+
+    Helper function called in create/update template version mutations
+    """
+    # Convert graphql relay ids to primary keys
+    study_ids = []
+    for sid in study_node_ids:
+        _, study_id = from_global_id(sid)
+        study_ids.append(study_id)
+
+    # Check all studies exist
+    studies = Study.objects.filter(pk__in=study_ids).only("kf_id").all()
+    if len(studies) != len(study_ids):
+        raise GraphQLError(
+            "Failed to create/update template version because one or more "
+            "studies in the input do not exist. "
+        )
+    return studies
+
+
+class CreateTemplateVersionInput(graphene.InputObjectType):
```

Comment:
are DataTemplates and Templates the same thing?

Reply:
Do you mean are `DataTemplates` and `TemplateVersions` the same thing? If yes, then my answer is no they are different concepts. See https://github.com/kids-first/kf-api-study-creator/blob/templates-rfc/docs/rfcs/0002_templates_mvp.md#datatemplate

But now I'm thinking more about it...I guess the `DataTemplate` is really just a container for different versions of a template's field definitions. So maybe I could have called `TemplateVersion` -> `FieldDefinitions`?

Reply:
I mean is DataTemplate like a sub type of Template? If they are the same I think we should use either DataTemplate or Template everywhere. `FieldDefinitions` would also solve this but you might get some weird types with the `field_definitions` field under a `FieldDefinitions` object.

Reply:
What is `Template`? I don't think we have that concept anywhere so I don't know what a subtype of Template means..I want to avoid calling these templates, `Template,` bc I think that is too general. These are data submission templates, and there could be templates for other things too in the future. `DataTemplate` is kind of general too but I thought `DataSubmissionTemplate` was too verbose

Reply:
you have the concept right here: `CreateTemplateVersionMutation` 
if the actual model is `DataTemplate` then should this be `CreateDataTemplateVersionMutation`?

Reply:
Ohh I see what you're saying.. so instead of naming the model `TemplateVersion` we would name it `DataTemplateVersion`?

Reply:
i think the confusion here is calling them `DataTemplates` and then calling the versions `TemplateVersions`

Reply:
Ok, if everyone thinks `DataTemplateVersion` makes things less confusing, I can change `TemplateVersion` to `DataTemplateVersion`. React thumbs up if you want this change!

Reply:
I left it as `TemplateVersion` in favor of brevity